### PR TITLE
Add `aug_decay` scheduler for RT-DETR YOLOv8 augmentations

### PR DESCRIPTION
**Pull request by artest08**
Adds an `aug_decay` training flag that linearly decays YOLOv8-style augmentations (`mosaic`, `mixup`, `cutmix`, `copy_paste`) toward zero across RT-DETR training and zeros the remaining spatial and color augmentations for the final 4 epochs. When enabled, `close_mosaic` is set to 0 so the decay schedule owns the clean tail rather than the base trainer's hard close.

```python
from ultralytics import RTDETR

RTDETR("rtdetr-l.pt").train(data="coco8.yaml", epochs=72, aug_decay=True)
```

## 🛠️ PR Summary

<sub>Made with ❤️ by [Ultralytics Actions](https://www.ultralytics.com/actions)</sub>

### 🌟 Summary
✨ Adds optional scheduled decay of YOLOv8-style augmentations during RT-DETR training, ending with a clean four-epoch training phase.

### 📊 Key Changes
- Introduces the `aug_decay` training option in the default configuration and CLI argument list.
- Adds an RT-DETR dataset scheduler that gradually reduces `mosaic`, `mixup`, `cutmix`, and `copy_paste` probabilities.
- Disables remaining image augmentations during the final four training epochs.
- Updates dataset transforms at the start of each epoch and resets the training loader when needed.
- Prevents the standard hard-close mosaic behavior from overriding the new schedule.
- Documents the internal `_compute_scheduled_prob` helper in the RT-DETR validation reference.

### 🎯 Purpose & Impact
- 🎯 Provides a smoother transition from heavily augmented data to clean images during RT-DETR training.
- 📈 May improve final model convergence and accuracy by allowing the model to refine predictions on unaugmented examples.
- ⚙️ Keeps existing behavior unchanged by default because `aug_decay` is disabled unless explicitly enabled.
- 🧩 Gives users finer control over RT-DETR augmentation scheduling without affecting other model families.

### CODE REVIEW

**Comment by UltralyticsAssistant** [comment]
👋 Hello @artest08, thank you for submitting a `ultralytics/ultralytics` 🚀 PR! This is an automated message to help with review readiness—an engineer will assist you shortly. Please review the checklist below for smooth integration 😊

- ✅ **Define a Purpose**: Clearly explain the purpose of your fix or feature in your PR description, and link to any [relevant issues](https://github.com/ultralytics/ultralytics/issues). Ensure your commit messages are clear, concise, and adhere to the project's conventions.
- ✅ **Synchronize with Source**: Confirm your PR is synchronized with the `ultralytics/ultralytics` `main` branch. If it's behind, update it by clicking the 'Update branch' button or by running `git pull` and `git merge main` locally.
- ✅ **Ensure CI Checks Pass**: Verify all Ultralytics [Continuous Integration (CI)](https://docs.ultralytics.com/help/CI/) checks are passing. If any checks fail, please address the issues.
- ✅ **Update Documentation**: Update the relevant [documentation](https://docs.ultralytics.com/) for any new or modified features.
- ✅ **Add Tests**: If applicable, include or update tests to cover your changes, and confirm that all tests are passing.
- ✅ **Sign the CLA**: Please ensure you have signed our [Contributor License Agreement](https://docs.ultralytics.com/help/CLA/) if this is your first Ultralytics PR by writing "I have read the CLA Document and I sign the CLA" in a new message.
- ✅ **Minimize Changes**: Limit your changes to the **minimum** necessary for your bug fix or feature addition. _"It is not daily increase but daily decrease, hack away the unessential. The closer to the source, the less wastage there is."_  — Bruce Lee

For more guidance, please refer to our [Contributing Guide](https://docs.ultralytics.com/help/contributing/). Don't hesitate to leave a comment if you have any questions. Thank you for contributing to Ultralytics! 🚀

**Review by UltralyticsAssistant — APPROVED** [review]
## 🔍 PR Review

<sub>Made with ❤️ by [Ultralytics Actions](https://www.ultralytics.com/actions)</sub>

Looks clean overall. The new `aug_decay` wiring is consistent across config, trainer, and RT-DETR dataset transform rebuilding, and I didn't find any clear correctness or maintainability issues in the changed lines.

**Comment by glenn-jocher** [comment]
Thanks for the contribution — please keep this RT-DETR-specific and avoid adding a new global `default.yaml` arg unless it’s broadly reusable across models, then add a small test for the epoch-wise transform rebuild and final no-augmentation tail; the preferred extension point for this kind of scheduling is the [custom trainer guide](https://docs.ultralytics.com/guides/custom-trainer), since [RT-DETR dataset transforms](https://docs.ultralytics.com/reference/models/rtdetr/val) already centralize `build_transforms()`.

**Comment by glenn-jocher** [comment]
**Simplification suggestion (Core Principles: Delete > Replace > Add)** 🧹

A few behavior-preserving trims:

1. Inline `_compute_scheduled_prob` (single caller) into `_build_yolov8_aug_hyp` and drop it from the public reference docs — a `_`-prefixed private helper shouldn't be documented.
2. Delete the `_rtdetr_aug_callback_registered` guard — `train()` runs once per trainer, so double-registration can't happen.
3. Drop the `trainer=None` default + `trainer = trainer or self` fallback — `run_callbacks` always passes `self`.
4. `build_transforms(hyp=self.base_hyp)` passes an arg that's immediately overwritten when `aug_decay` is on — drop it.

Bigger picture for a maintainer call: the "clean tail" + per-epoch `train_loader.reset()` largely duplicates the existing `close_mosaic` flow. Only the linear decay is genuinely new — worth deciding whether this needs a new global cfg key at all vs. reusing `close_mosaic` for the tail. Also note: with `epochs <= 4`, `aug_decay` silently disables *all* augmentation.
